runtime/src/qio/qio_formatted.c

```diff
@@ -361,6 +361,86 @@ qioerr _append_char(char* restrict * restrict buf, size_t* restrict buf_len, siz
   return 0;
 }
 
+qioerr read_entire_file(const int threadsafe,qio_channel_t* restrict ch, const char* restrict* restrict out, int64_t* restrict len_out, ssize_t maxlen)
+{  
+  qioerr err,err_reading_file=0;
+  int64_t file_length;
+  char* ret=NULL;
+  ssize_t amt;
+  err_t errcode;
+  int64_t fileLenGuess=1024;
+  if( threadsafe ) 
+  {
+    err = qio_lock(&ch->lock);
+    if( err ) return err;
+  }
+
+  err = qio_channel_mark(false, ch);
+  if( err ) goto unlock;
```

Comment:
You shouldn't have these two lines, since we don't need to mark, and we removed the code to revert/commit.

Reply:
the locking is to protect the channel data structure

Reply:
OK